SplunkAppForWazuh/appserver/static/js/directives/wz-aws-bar/wz-aws-bar.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * Wazuh app - Filter bar directive
+ * Copyright (C) 2018 Wazuh, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Find more information about this on the LICENSE file.
+ */
+define(['../module'], function (directives) {
+  'use strict'
+  directives.directive('wazuhAwsBar', function ($notificationService, BASE_URL) {
+    return {
+      restrict: 'E',
+      controller: function ($scope, $currentDataService) {
+        /**
+         * Prettifies filters for md-chips
+         * @returns {Array}
+         */
+        const getPrettyFilters = () => {
+          let prettyFilters = []
+          const uglyFilters = $currentDataService.getFilters()
+          if (uglyFilters && uglyFilters.length > 0) {
+            for (const filter of uglyFilters) {
+              const key = Object.keys(filter)[0]
+              prettyFilters.push(`${key}:${filter[key]}`)
+            }
+          }
+          const awsUglyFilters = JSON.parse(window.localStorage.getItem('awsSourceFilters'))
```

Comment:
Do not use Ugly as a variable name. Don't use JSON.parse and similar functions without try/catch